lib/zip_source_zip_new.c

```diff
@@ -69,6 +92,13 @@ zip_source_t *_zip_source_zip_new(zip_t *srcza, zip_uint64_t srcidx, zip_flags_t
         return NULL;
     }
 
+    /* stat may come from third party code, we need to validate it */
+    if ((st.valid & ZIP_STAT_SIZE) == 0) {
+        int ze = (reopening ? ZIP_ER_INVAL : ZIP_ER_INTERNAL);
+        zip_error_set(error, ze, 0);
+        return NULL;
+    }
```

Comment:
It's not an error if we don't know the size, we'll just have to deal with it. Some checks won't be possible later on, but that's fine.